### ISSUE TITLE
Fwupd (1) has wrong icon, (2) does not launch from the manager, (3) gives error when run from terminal. EDIT: does ANY termina app launch properly?

### ISSUE THREAD

**Issue by LinuxOnTheDesktop** [linuxmint/mintinstall#249]
I see this in the Software Manager:

![image](https://user-images.githubusercontent.com/47056543/57741835-79236900-76b6-11e9-8d7e-381e375e9fcc.png)

1) Is that .deb icon right? I am not installing a .deb?
2) After installation, the Launch button does nothing.
3) From the terminal, I see this:

```
$  sudo flatpak run org.freedesktop.fwupd/x86_64/stable get-devices
flatpak run: Can't find a11y bus: GDBus.Error:org.freedesktop.DBus.Error.TimedOut: Failed to activate service 'org.a11y.Bus': timed out (service_start_timeout=120000ms)
Loading…                 [-                                      ]01:02:48:0865 FuPluginUefi         Unable to determine EFI system partition location, override using OverrideESPMountPoint in uefi.conf
Loading… [. . .]
```
Linux Mint 19.1 x64 Cinnamon. (I am using the alpha but I had the same sort of problem earlier, without the alpha version of Mint.)

I resort to a flatpak, with its seemingly very cumbersome way of launching an app (is there no easier way to run a flatpak app?),  because:

- the repo version of `fwupd` (or `fwupdmgr` or whatever it is called) produces errors, which the fwupd devs say they cannot troubleshoot because the sofrware is to old;
- I had trouble getting the snap version installed;
- I cannot build the software from source because the build tools in the repos are too old.

**Comment by LinuxOnTheDesktop** [comment]
Indeed: does **any** _terminal_ program launch properly directly from MintInstall?

**Comment by xenopeek** [comment]
The wrong icon could be due the icon theme you have configured. Or it could be due to the less advanced flatpak repo handling that is in the 2.5 year older mintinstall you're running. I don't have that version to confirm. Testing this on Linux Mint 20.1 and searching for "firmware" in the flatpak category it shows correct icons:
![image](https://user-images.githubusercontent.com/21159972/117582329-cb5ebf80-b101-11eb-918f-112d5323eee6.png)

As for why fwupd flatpak won't run that is not a mintinstall issue. For help with troubleshooting such problems please make use of the support forums.

**Comment by LinuxOnTheDesktop** [comment]
Thanks. But the fwupd flatpak not running *might* be a mintInstall issue. Indeed - and see my edit of this page's title - one may wonder whether mintInstall is able to run any terminal application.

**Comment by xenopeek** [comment]
Before reporting issues please retest on the current release. To avoid discussing something that may already have been addressed in past 2.5 years.

To answer the current mintinstall looks for a .desktop file, looks for a /usr/bin/_packagename_ file or in case of flatpak just runs the flatpak. See this section of the code: https://github.com/linuxmint/mintinstall/blob/32b2eeb58f13c7f6dc29eddc11699aec27a88ac6/usr/lib/linuxmint/mintinstall/mintinstall.py#L2305
fwupd, being a terminal program, doesn't have a .desktop file (menu launcher) and the binary name (fwupdtool) is different from the package name (fwupd) so mintinstall won't find how to run it. In case of fwupd installed as flatpak that doesn't matter as flatpak run command takes care of that. But if you run fwupdtool that doesn't take user input, it just shows help and exits—closing the terminal window. fwupdtool requires you to pass it arguments to do something useful. Run it from the terminal.
